apps/worker/src/app/workflow/usecases/send-message/send-message-push.usecase.ts

```diff
@@ -113,81 +120,103 @@ export class SendMessagePush extends SendMessageBase {
         Object.values(PushProviderIdEnum).includes(chan.providerId as PushProviderIdEnum)
       ) || [];
 
-    const messagePayload = Object.assign({}, command.payload);
-    delete messagePayload.attachments;
-
     if (!pushChannels.length) {
-      await this.sendErrors(command);
+      await this.sendNoActiveChannelError(command.job);
+      await this.sendNotificationError(command.job);
```

Comment:
I forgot to raise this before, I am not sure if we need to use the word `notification` as it is in NOTIFICATION_ERROR because we are in the `job` level here and not in the notification level, the notification should cover all of the workflow.

regarding the comment:
If we want to add a summary we should add it to the run-job use case in its catch block so we would add a summary to all of the potential cases, but I am not sure it is necessary addition to that we do not add there any data besides that there was an undefined errors. let me know what you think about it.

Reply:
> I forgot to raise this before, I am not sure if we need to use the word `notification` as it is in NOTIFICATION_ERROR because we are in the `job` level here and not in the notification level, the notification should cover all of the workflow.
> 
I agree with you in the codebase but UI wise for the user what it fails is the notification for a certain subscriber (or notifications as the Push is being a different case where multiple providers can be in play). I can change the DetailEnum for this case for something more correct codewise though for the UI I find proper to use notification.
Any suggestion for the DetailEnum?


 
> regarding the comment: If we want to add a summary we should add it to the run-job use case in its catch block so we would add a summary to all of the potential cases, but I am not sure it is necessary addition to that we do not add there any data besides that there was an undefined errors. let me know what you think about it.

Currently Activity Feed takes the last execution details to show in the summary for the card for every subscriber and notification for all channels. The different flow of Push made me care about if something fails to have that execution details to tell "Hey look inside something was wrong" without having to think in all potential combinations.

It is true that for the case of NoActiveChannelError might not be needed, but I tried to be consistent for the behaviour of Push in the ActivityFeed. I can change it making `sendNoActiveChannelError` to throw and removing `sendNotificationError` if you think it is better for the user awareness in the UI. 

I would avoid moving that responsibility to RunJob as that triggers the different channels in different ways (email and SMS has single provider per node with multi provider store, chat and push have no multi provider store but can in the same node multiple active providers).

Reply:
> I agree with you in the codebase but UI wise for the user what it fails is the notification for a certain subscriber (or notifications as the Push is being a different case where multiple providers can be in play). I can change the DetailEnum for this case for something more correct codewise though for the UI I find proper to use notification.
Any suggestion for the DetailEnum?

I see the reasoning there now, thanks for the explanation. i think we can keep it like this.

> Currently Activity Feed takes the last execution details to show in the summary for the card for every subscriber and notification for all channels. The different flow of Push made me care about if something fails to have that execution details to tell "Hey look inside something was wrong" without having to think in all potential combinations.

If the concern is in the UI maybe we could add this message generated in the UI based on the data. wdyt?

> It is true that for the case of NoActiveChannelError might not be needed, but I tried to be consistent for the behaviour of Push in the ActivityFeed. I can change it making sendNoActiveChannelError to throw and removing sendNotificationError if you think it is better for the user awareness in the UI.

the NoActiveChannelError is needed and provides more information, but the addition sendNotificationError does not provide additional information to the debugging.